scope_registry.go

```diff
@@ -21,64 +21,89 @@
 package tally
 
 import (
+	"hash/maphash"
+	"runtime"
 	"sync"
 	"unsafe"
 )
 
 var scopeRegistryKey = keyForPrefixedStringMaps
 
 type scopeRegistry struct {
```

Comment:
I could do that, but that would affect performance because you are suggesting that a Get() is going to compute the bucket hash and then the hash for the internal map.

So what of the function Subscope which does multiple gets and multiple sets? We would be repeating the bucket hash many times when we don't need to.

Although I could create a function that returns the pointer to the subscope, but I don't feel that is clean to have a public api to return an internal pointer.